01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md

```diff
@@ -0,0 +1,607 @@
+terminal application called Windows Terminal that is available for Windows 10+
+users.
+
+The Windows Terminal is a new, modern, fast, efficient, powerful, and productive
+terminal application for users of command-line tools and shells like Command
+Prompt, PowerShell, and WSL.
+
+You can download it from the Microsoft Store:
+[Windows Terminal](https://www.microsoft.com/p/windows-terminal/9n0dx20hk701) or
+from the GitHub releases page:
+[Windows Terminal Releases](https://github.com/microsoft/terminal/releases)
+(download the `.msixbundle` file).
+
+Install it and open it.
+
+### Configure Windows Terminal
+
+By default, Windows Terminal will open PowerShell. Now that you have installed
+Windows Terminal, you can configure it to open WSL by default.
+
+Access the settings by clicking on the down arrow in the title bar and selecting
+"Settings". Set the two following settings:
+
+- **Default profile**: Ubuntu
+- **Default terminal application**: Windows Terminal
+
+Press **Save** to save the settings.
+
+![Configure Windows Terminal](./images/windows-terminal-configure.png)
+
+Close Windows Terminal and open it again. It should now open Ubuntu by default.
+
+You now have a modern terminal application that supports multiple tabs, multiple
+shells, and many other features.
+
+You now have access to a Linux distribution on your Windows machine. You can use
+it to run Linux commands and utilities on Windows.
+
+All future commands in this guide and course will be run in the Ubuntu terminal
+unless specified otherwise.
+
+You can always open a PowerShell terminal by clicking on the down arrow in the
+title bar and selecting "Windows PowerShell".
+
+You can run a PowerShell terminal as an administrator by clicking on the down
+arrow in the title bar and selecting "Windows PowerShell" while holding the
+`Ctrl` key.
+
+![Open PowerShell](./images/windows-terminal-open-powershell.png)
+
+## Update Ubuntu
+
+Before starting the installation of the development environment, make sure that
+your Ubuntu installation is up to date. You can update Ubuntu by running the
+following commands in the terminal. You might need to enter your password when
+running the commands:
+
+> [!TIP]
+>
+> The `sudo` command is used to run commands with superuser privileges. You will
+> be prompted to enter your password when running a command with `sudo`.
+>
+> This is a security feature of Linux to prevent unauthorized access to the
+> system.
+
+> [!TIP]
+>
+> All Linux distributions come with a package manager that is used to install,
+> update, and remove packages. The package manager for Ubuntu is called `apt`.
+>
+> The package manager uses a package list to know which packages are available
+> for installation. You can then install any package from the package list using
+> the package manager.
+>
+> If you are interested to have a package manager for Windows, you can check
+> [Chocolatey](https://chocolatey.org/) or [Scoop](https://scoop.sh/).
```

Comment:
I would also add [winget](https://learn.microsoft.com/fr-fr/windows/package-manager/) , as it is the package management solution created by Microsoft.

Reply:
Good point, I have added it in my last commit.